robotools/evotools/__init__.py

```diff
@@ -181,6 +181,369 @@ def _int_to_tip(tip_int: int):
+    cleaner_location : tuple
+        Tuple with grid position (1-67) and site number (0-127) of cleaner as integers
+    arm : int
+        number of the LiHa performing the action: 0 = LiHa 1, 1 = LiHa 2
+    waste_vol: float
+        Volume in waste in mL (0-100)
+    waste_delay : int
+        Delay before closing valves in waste in ms (0-1000)
+    cleaner_vol: float
+        Volume in cleaner in mL (0-100)
+    cleaner_delay : int
+        Delay before closing valves in cleaner in ms (0-1000)
+    airgap : float
+        Volume of airgap in µL which is aspirated after washing the tips (system trailing airgap) (0-100)
+    airgap_speed : int
+        Speed of airgap aspiration in µL/s (1-1000)
+    retract_speed : int
+        Retract speed in mm/s (1-100)
+    fastwash : int
+        Use fast-wash module = 1, don't use it = 0
+    low_volume : int
+        Use pinch valves = 1, don't use them = 0
+
+    Returns
+    -------
+    tips : list
+        Tip(s) that will be selected; have been converted to tip.T1 - tip.T8 here if they weren't originally formatted that way
+    waste_location : tuple
+        Tuple with grid position (1-67) and site number (0-127) of waste as integers
+    cleaner_location : tuple
+        Tuple with grid position (1-67) and site number (0-127) of cleaner as integers
+    arm : int
+        number of the LiHa performing the action: 0 = LiHa 1, 1 = LiHa 2
+    waste_vol: float
+        Volume in waste in mL (0-100)
+    waste_delay : int
+        Delay before closing valves in waste in ms (0-1000)
+    cleaner_vol: float
+        Volume in cleaner in mL (0-100)
+    cleaner_delay : int
+        Delay before closing valves in cleaner in ms (0-1000)
+    airgap : float
+        Volume of airgap in µL which is aspirated after washing the tips (system trailing airgap) (0-100)
+    airgap_speed : int
+        Speed of airgap aspiration in µL/s (1-1000)
+    retract_speed : int
+        Retract speed in mm/s (1-100)
+    fastwash : int
+        Use fast-wash module = 1, don't use it = 0
+    low_volume : int
+        Use pinch valves = 1, don't use them = 0
+    """
+    if tips is None:
+        raise ValueError("Missing required parameter: tips")
+
+    def _int_to_tip(tip_int: int):
+        """Asserts a Tecan Tip class to an int between 1 and 8."""
+        if not 1 <= tip_int <= 8:
+            raise ValueError(
+                f"Tip is {tip} with type {type(tip)}, but should be an int between 1 and 8 for _int_to_tip conversion."
+            )
+        if tip_int == 1:
+            return Tip.T1
+        elif tip_int == 2:
+            return Tip.T2
+        elif tip_int == 3:
+            return Tip.T3
+        elif tip_int == 4:
+            return Tip.T4
+        elif tip_int == 5:
+            return Tip.T5
+        elif tip_int == 6:
+            return Tip.T6
+        elif tip_int == 7:
+            return Tip.T7
+        elif tip_int == 8:
+            return Tip.T8
```

Comment:
This is the version that should be extracted & used in both cases (see comment above)